learn/tooling-guide/ballerina-shell.md

```diff
@@ -2,11 +2,197 @@
 layout: ballerina-tooling-guide-left-nav-pages-swanlake
 title: Ballerina Shell
 permalink: /learn/tooling-guide/ballerina-shell/
+description: The Ballerina Shell is a Read-Evaluate-Print Loop (REPL) for Ballerina.
 active: ballerina-shell
-intro: The VS Code Ballerina extension gives you the same debugging experience as the conventional VS Code Debugger. Thus, you can run or debug your Ballerina programs easily via the VS Code Ballerina extension by launching its debugger. 
+intro: The Ballerina Shell is a Read-Evaluate-Print Loop (REPL) for Ballerina.
 redirect_from:
   - /learn/tooling-guide/ballerina-shell
 ---
 
+It allows you to evaluate snippets of code without having to write complete programs. 
 
+This will particularly be your interest if you are looking to try out the language and its new features, and also for debugging purposes.
 
+## Setting up the Prerequisites
+
+The Ballerina Shell is a part of the Ballerina distribution. It was first shipped with Ballerina Swan Lake Alpha1. Therefore, if you are using Swan Lake Alpha1 or a later version, you are all set. If not, for information on installing the latest version of Ballerina, see [downloads](https://ballerina.io/downloads/).
+
+## Starting the Shell
+
+Execute the command below to start Ballerina Shell.
+
+```bash
+$ bal shell
+Welcome to Ballerina Shell REPL.
+Type /exit to exit and /help to list available commands.
+
+=$ 
+```
+
+## Stopping the Shell
+
+Execute the `/exit` command below in a new prompt to exit the Shell.
+
+```bash
+=$ /exit                                                                                                                                                                                                           
+| Bye!!!
+```
+
+## Evaluating Code Snippets
+
+The types of code snippets below are supported in the Shell.
+
+- Import declarations
+- Module-level declarations (except services)
+- Statements
+- Expressions
+
+### Importing a Module
+
+A module can be imported as follows. Once a module is imported, it can be used in any subsequent snippet run in that particular Shell session. This is true for other module-level declarations as well.
+
+```bash
+=$ import ballerina/io
+```
+
+Execute the `/imports` command below to view the list of the imported modules.
```

Comment:
```suggestion
Execute the `/imports` command to view the list of the imported modules.
```

Reply:
Same comment as above.

Reply:
This is a different case. This is not referring to the code snippet. The code snippet is there just as an illustrative example of what's described here.

Reply:
Got it. Then, shall we say: "Execute the `/imports` command to view the list of the imported modules as shown below."/

Reply:
Redundant isn't it? 🤔